Hashtable.c

```diff
@@ -186,11 +254,9 @@ static void insert(Hashtable* this, ht_key_t key, void* value) {
 
 void Hashtable_setSize(Hashtable* this, size_t size) {
 
+   assert(size == 0 || size > this->items);
    assert(Hashtable_isConsistent(this));
 
-   if (size <= this->items)
-      return;
-
    size_t newSize = nextPrime(size);
```

Comment:
For `size==0` this resizes the Hashtable below `this->items` when more then `MIN_TABLE_SIZE` items are currently present.

Reply:
I don't believe it's possible to call this with a size less than `MIN_TABLE_SIZE` since we check for that in [`Hashtable_shouldResize`](https://github.com/htop-dev/htop/blob/bb0dd3f2292cd6e9ecf407c251f0e5f46f83794b/Hashtable.c#L167-L172). Is there something I missed?

https://github.com/htop-dev/htop/blob/bb0dd3f2292cd6e9ecf407c251f0e5f46f83794b/Hashtable.c#L167-L172

Reply:
Yes: `Hashtable_setSize` is public in the header and thus this additional `size == 0` check is reachable …

Reply:
Good catch and fixed.

Reply:
> Actually trying to run the code produces an infinite loop in Hashtable_put because the resizing doesn't trigger. Can you have another look? [comment](https://github.com/htop-dev/htop/pull/998#issuecomment-1119281586)

Checking if `size == 0` here is what lead to this issue. This happens in `Hashtable_put` is called after all of its elements have been removed (so it attempts to call `Hashtable_setSize` with a `size` of 0. IMO, calling `Hashtable_setSize` with `size == 0` because we round it up to the next prime, which is `MIN_TABLE_SIZE`.

I changed `Hashtable_setSize` to check the size after rounding up to the next nearest prime and to assert that the rounded up size is greater than the current number of items.

https://github.com/htop-dev/htop/blob/c9c6fc9a7a04d75d2251d3ed38940ede03cf479d/Hashtable.c#L259-L263